plugins/Gravatar/Gravatar.php

```diff
@@ -104,24 +104,14 @@ function config() {
 	 * Register event hooks for plugin.
 	 */
 	function hooks() {
-		return array(
-			'EVENT_USER_AVATAR' => 'user_get_avatar',
-			'EVENT_LAYOUT_RESOURCES' => 'csp_headers',
-		);
-	}
-
-	/**
-	 * Add Content-Security-Policy for retrieving Avatar images.
-	 *
-	 * @return void
-	 */
-	function csp_headers() {
-		# Policy for images: Allow gravatar URL
 		if( config_get( 'show_avatar' ) !== OFF ) {
 			# Set CSP header
-			header( "Content-Security-Policy: img-src 'self' " .
-				self::getAvatarUrl() );
+			http_csp_add( 'img-src', self::getAvatarUrl() );
```

Comment:
Sorry i can't provide much feedback here, since i am no expert. But, as i understand, the treatment of CSP directives probably fits in it's own functionality, i.e, separated from other header declarations.

- Is the needed action, for plugins, always to "add"?
i found that sometines, i tried to do something that gets blocked by mantis CSP defaults. In those cases, do i need to add a new directive, or do i actually need to "remove" one of the mantis ones?